### PR TITLE
Ensure that when merged to master, images are tagged and pushed with new version to Docker hub

### PR DESCRIPTION
**Pull request by hedss**
Connects-to: #284
Change-type: patch
---- Autogenerated Waffleboard Connection: Connects to #284

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -34,3 +34,11 @@ deployment:
           docker push ${DOCKER_IMAGE}:${CIRCLE_TAG}
         fi
 
+        # On master merges, ensure we tag the new version.
+        [[ "${CIRCLE_BRANCH}" == "master" ]] || exit 0
```
Comment (brownjohnf) [inline circle.yml]: Is there a reason you don't want to run this on other branches? It'll generate a nice tag like v2.3.0-<sha>, which allows one to pull/run builds from branches.

Comment (hedss) [reply circle.yml]: That's a good point. I'll remove the `master` only check.

**Comment by hedss** [comment]
@lekkas Yep, `git` is actually already included, so didn't need the line anyway!

**Comment by brownjohnf** [comment]
I think the reason we have to add it to some of our jobs is that we don't have it when we use a custom build env image on circle, like when we need docker-in-docker

**Pull request merged**